pydata_google_auth/auth.py

```diff
@@ -248,12 +265,33 @@ def get_user_credentials(
         "installed": {
             "client_id": client_id,
             "client_secret": client_secret,
-            "redirect_uris": ["urn:ietf:wg:oauth:2.0:oob"],
+            "redirect_uris": [redirect_uri, "urn:ietf:wg:oauth:2.0:oob"],
             "auth_uri": GOOGLE_AUTH_URI,
             "token_uri": GOOGLE_TOKEN_URI,
         }
     }
 
+    def run_webapp(self,
```

Comment:
I think this might be slightly easier to follow if we rename "self" to "flow" or something similar. AFAIK, the name "self" is only a convention, and even if we're monkeypatching, I think a different name would be easier to follow.

Also, if we're passing in `redirect_uri` does this actually need to be a closure function? Could we move this to a private method of the `auth` module?

Reply:
Done.